core/test/concatenate_test.cpp

```diff
@@ -211,3 +211,22 @@ TEST(ConcatenateTest, concatenate_sparse_no_data) {
   EXPECT_EQ(y.coords()[Dim::X], concatenate(var1, var2, Dim::Y));
   EXPECT_EQ(y.labels()["labs"], concatenate(var1, var2, Dim::Y));
 }
+
+TEST(ConcatenateTest, dataset_with_no_data_items) {
+  Dataset a, b;
+  a.setCoord(Dim::X,
+             makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{1, 2}));
+  a.setLabels("abc",
+              makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{.1, .2}));
+  b.setCoord(Dim::X,
+             makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{3, 4}));
+  b.setLabels("abc",
+              makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{.3, .4}));
+
+  const auto res = concatenate(a, b, Dim::X);
+
+  EXPECT_EQ(res.coords()[Dim::X],
+            makeVariable<double>(Dims{Dim::X}, Shape{4}, Values{1, 2, 3, 4}));
+  EXPECT_EQ(res.labels()["abc"], makeVariable<double>(Dims{Dim::X}, Shape{4},
+                                                      Values{.1, .2, .3, .4}));
+}
```

Comment:
Can you add a test with an additional coord or label that represents bin edges?

Reply:
Without an associated data item how can you determine if a coordinate represents bin edges or not?

Reply:
It looks at the dims of the other coords or labels.